clicommand/env.go

```diff
@@ -34,38 +35,27 @@ var EnvCommand = cli.Command{
 		},
 	},
 	Action: func(c *cli.Context) error {
-		env := os.Environ()
-		envMap := make(map[string]string, len(env))
-
-		for _, e := range env {
-			k, v, _ := strings.Cut(e, "=")
+		envn := os.Environ()
+		envMap := make(map[string]string, len(envn))
+
+		for _, e := range envn {
+			k, v, ok := env.Split(e)
+			if !ok {
+				fmt.Fprintf(c.App.ErrWriter, "Invalid environment variable from os.Environ: %q\n", e)
+				os.Exit(2)
+			}
```

Comment:
we're changing the failure dynamics here - previously, we'd ignore any envar that we weren't able to parse, and now we'll fail the command.

this change is all good, i'm also keen to make sure that this it's here intentionally, and isn't just an artifact of debugging or something

Reply:
I'm not sure why I was feeling paranoid about it here. os.Environ has to produce well-formed strings and the only way it wouldn't is if things are severely broken on the machine... but then all bets are off anyway.